bot/exts/moderation/incidents.py

```diff
@@ -404,9 +460,41 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
     async def on_message(self, message: discord.Message) -> None:
         """Pass `message` to `add_signals` if and only if it satisfies `is_incident`."""
         if is_incident(message):
+            message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
+            print(message_links)
+            if message_links:
+
+                embeds = []
+                for message_link in message_links:
+                    ctx = await self.bot.get_context(message)
+                    embeds.append(await make_message_link_embed(ctx, message_link))
+
+                try:
+                    webhook = await self.bot.fetch_webhook(Webhooks.incidents)
+                    webhook_embed_list = list(grouper(embeds, 10))
+                    webhook_msg_ids = []
+
+                    for x, embed in enumerate(webhook_embed_list):
+                        webhook_msg = await webhook.send(
+                            embeds=[x for x in embed if x is not None],
+                            username=sub_clyde(message.author.name),
+                            avatar_url=message.author.avatar_url,
+                            wait=True
+                        )
+                        webhook_msg_ids.append(webhook_msg.id)
+                        log.trace(f"Message Link Embed {x + 1}/{len(webhook_embed_list)} Sent Succesfully")
```

Comment:
The try clause should be minimized to the part you think will throw an exception, e.g without the part before the loop.

I also suggest you export the loop to a utility function.